spec/chatgpt.spec.ts

```diff
@@ -73,14 +92,34 @@ describe("ChatGPT Flow", () => {
     const request = createRequest(TestPrompt, [TEST_WEATHER_FUNC_SCHEMA]);
     flow.logRequest(request);
 
+    let StartTime = new Date().toISOString();
+
     const response = await openai.chat.completions.create(request);
-    flow.logResponse(response);
 
-    if (response.choices[0]?.message?.function_call) {
-      flow.logFunctionCall(response.choices[0]?.message?.function_call);
+    let EndTime = new Date().toISOString();
+
+    flow.logResponse({ ...response, start_time: StartTime, end_time: EndTime });
+
+    let functionCallStartTime;
+    let functionCallEndTime;
+
+    console.log(response.choices[0]?.message.tool_calls);
+
+    if (response.choices[0]?.message?.tool_calls) {
+      const functionCall =
+        response.choices[0]?.message?.tool_calls[0]?.function;
+      const args = JSON.parse(functionCall.arguments);
+      functionCallStartTime = new Date().toISOString();
+      const weatherData = TEST_WEATHER_FUNC_IMPL(args.location);
+      functionCallEndTime = new Date().toISOString();
+      flow.logFunctionCall({
```

Comment:
What I am parsing is the arguments, not the functionCall object, so that I can call function and return the result.

Or do you want me to do this while processing?